analog_forecast.py

```diff
@@ -0,0 +1,297 @@
+        
+    Returns:
+        analogs (xarray.DataArray): data array of RMSE values and dates for 5 best analogs
+    """
+    # compute RMSE between ref_date and all preceding dates 
+    #  for the specified variable and spatial domain
+    rmse_da = run_rmse_over_time(da, ref_date, "any")
+    varname = da.name
+    
+    # subset to first 5 analogs for now
+    # impose restrictions on temporal proximity for analog based on variable.
+    #  if SST, then we use an exclusion buffer of 6 months, atmospheric vars use 30 days
+    if varname in ["t2m", "msl", "z"]:
+        buffer = 30
+    elif varname in ["sst"]:
+        buffer = 180
+    analogs = take_analogs(rmse_da, buffer, ref_date, 5)
+    
+    if print_analogs:
+        print("   Top 5 Analogs: ")
+        for rank, date, rmse in zip(
+            [1, 2, 3, 4, 5], pd.to_datetime(analogs.time.values), analogs.values
+        ):
+            print(f"Rank {rank}:   Date: {date:%Y-%m-%d};  RMSE: {round(rmse, 3):.3f}")
+    
+    assert ~any(np.isnan(analogs))
+    
+    return analogs
+
+
+def make_forecast(sub_da, times, ref_date):
+    """Use a dataarray of analogs containing dates to create a composite forecast for 14 days following reference date
+    
+    Args:
+        sub_da (xarray.DataArray): data array of ERA5 data for variable and region of interest
+        times (list): list of times to generate forecast from
+        ref_date (str): reference date in formate YYYY-mm-dd
+    
+    Returns:
+        forecast (xarray.DataArray): forecast values computed for the 14 days following ref_date. Computed as the mean of corresponding days following each analog. 
+    """
+    arr = []
+    for t in times:
+        time_sl = slice(
+            t + pd.to_timedelta(1, unit="d"),
+            t + pd.to_timedelta(14, unit="d")
+        )
+        arr.append(sub_da.sel(time=time_sl).values)
+
+    # take mean over axis 0 (year/analog axis) to get the forecast
+    #  will have shape of (14, n_lat, n_lon)
+    composite = np.array(arr).mean(axis=0)
+    
+    # construct a new data array
+    time = pd.date_range(
+        pd.to_datetime(ref_date + " 12:00:00") + pd.to_timedelta(1, unit="d"),
+        periods=14
+    )
+    forecast = xr.DataArray(
+        data=composite,
+        dims=sub_da.dims,
+        coords=dict(
+            longitude=(["longitude"], sub_da.longitude.data),
+            latitude=(["latitude"], sub_da.latitude.data),
+            time=time,
+        ),
+    )
+    
+    return forecast
+
+
+if __name__ == "__main__":
+    # parse some args
+    varname, ref_date, spatial_domain, workers = parse_args()
+    
+    # start dask cluster
+    client = Client(n_workers=workers, dashboard_address="localhost:33338")
+    # run analog search
+    # get the ERA5 data for searching
+    sub_da = read_subset_era5(spatial_domain, data_dir, varname, use_anom)
+    analogs = find_analogs(da, ref_date, print_analogs=True)
```

Comment:
shouldn't this line use the ```sub_da``` variable created above as input argument for the ```find_analogs``` function? Like this:

```analogs = find_analogs(sub_da, ref_date, print_analogs=True)```

Otherwise, I don't see where ```da``` variable comes from when run directly from the interpreter